### PR TITLE
Add computer name to azure sd

### PR DESCRIPTION
**Pull request by darshanime**
closes https://github.com/prometheus/prometheus/issues/8499
Signed-off-by: darshanime <deathbullet@gmail.com>

### CODE REVIEW

File: discovery/azure/azure.go [hunk 1]
```diff
@@ -490,6 +494,7 @@ func mapFromVMScaleSetVM(vm compute.VirtualMachineScaleSetVM, scaleSetName strin
 	return virtualMachine{
 		ID:                *(vm.ID),
 		Name:              *(vm.Name),
+		ComputerName:      *(vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName),
```
Comment (roidelapluie) [inline discovery/azure/azure.go:497]: Thanks for this. I tend to be very cautious  here and add a few nil checks. Are those fields all guaranteed to be here?

Thanks!

Comment (darshanime) [reply discovery/azure/azure.go]: thanks! addressed.

**Comment by roidelapluie** [comment]
Thanks!

File: discovery/azure/azure.go (hunk 1 above)
Comment (roidelapluie) [reply discovery/azure/azure.go]: we were not cautious enough :) https://github.com/prometheus/prometheus/issues/9382

Comment (darshanime) [reply discovery/azure/azure.go]: 😢 sorry about this. I saw you mentioned `I will work on a fix ASAP.` already, let me know if I can help.

**Pull request merged**